CLI: ping/sample not properly outputting some errors, like polling timeout

For example:
```
$ dwave ping --polling-timeout 0.00001 --json
{..., "error": ""}
```

The `error` field should contain a polling timeout error description.

As a result, `tests.test_cli.TestCliLive.test_ping_json_timeout_error` is also failing with:
```
======================================================================
FAIL: test_ping_json_timeout_error (tests.test_cli.TestCliLive)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/radomir/work/dwave-cloud-client/tests/test_cli.py", line 468, in test_ping_json_timeout_error
    self.assertEqual(result.exit_code, 9)
AssertionError: 127 != 9
```

--- a/dwave/cloud/cli.py
+++ b/dwave/cloud/cli.py
@@ -377,7 +377,7 @@ def _sample(solver, problem, params, output):
         response = solver.sample_ising(*problem, **params)
         problem_id = response.wait_id()
         output("Submitted problem ID: {problem_id}", problem_id=problem_id)
-        response.wait()
+        response.result()
     except RequestTimeout:
         raise CLIError("API connection timed out.", 8)
     except PollingTimeout:
@@ -424,7 +424,7 @@ def standardized_output(fn):
             output("Error: {error} (code: {code})", error=str(error), code=error.code)
             sys.exit(error.code)
         except Exception as error:
-            output("Unhandled error: {error}", error=str(error))
+            output("Unhandled error: {error}", error=repr(error))
             sys.exit(127)
         finally:
             flush()

--- a/releasenotes/notes/fix-ping-error-output-e83cead86e72d70d.yaml
+++ b/releasenotes/notes/fix-ping-error-output-e83cead86e72d70d.yaml
@@ -1,0 +1,6 @@
+---
+fixes:
+  - |
+    Preemptively resolve sampling answer during `dwave ping` and `dwave sample`,
+    as to capture and properly output/format eventual errors.
+    See `#517 <https://github.com/dwavesystems/dwave-cloud-client/issues/517>`_.

--- a/tests/api/resources/test_problems.py
+++ b/tests/api/resources/test_problems.py
@@ -81,7 +81,11 @@ class ProblemResourcesBaseTests(abc.ABC):
         p = ps.pop()
         self.assertIsInstance(p, models.ProblemStatus)
 
-    def test_list_filter(self):
+    # XXX: disable filter tests because the feature is currently broken on SAPI!
+    # TODO: enable as soon as the fix is deployed to SAPI prod. See #516.
+    # note: we don't actually use filtering anywhere in client's "public" API,
+    #       so the feature is not critical.
+    def disabled_test_list_filter(self):
         """Problem filtering works (on edges as well)."""
 
         # 404 on non-existing problem id

